### PR TITLE
Intake Subsystem Updates

### PR DESCRIPTION
**Pull request by FRC3484**
PR to capture feedback on intake subsystem.

### CODE REVIEW

File: 2024Robot_Team3484/src/main/cpp/subsystems/IntakeSubsystem.cpp
Comment (Alex-BusNet) [inline 2024Robot_Team3484/src/main/cpp/subsystems/IntakeSubsystem.cpp:12]: Pass CAN ID values through the constructor. This is more of proper programming practice in general instead of referencing the constants in the header file.
```
IntakeSubsystem::IntakeSubsystem(int ID_Pivot, int ID_Drive) : 
    _pivot_motor(ID_Pivot), _drive_motor(ID_Drive)
{

}
```

File: 2024Robot_Team3484/src/main/cpp/subsystems/IntakeSubsystem.cpp
Comment (Alex-BusNet) [inline 2024Robot_Team3484/src/main/cpp/subsystems/IntakeSubsystem.cpp:27]: I suggest you convert all of these hardcoded values in to constants (`#define`) in `Constants.h`

File: 2024Robot_Team3484/src/main/include/commands/auton/AutonIntakeCommand.h
Comment (Alex-BusNet) [inline 2024Robot_Team3484/src/main/include/commands/auton/AutonIntakeCommand.h:15]: Autonomous commands don't need the OI.

File: 2024Robot_Team3484/src/main/include/subsystems/IntakeSubsystem.h
Comment (Alex-BusNet) [inline 2024Robot_Team3484/src/main/include/subsystems/IntakeSubsystem.h:32]: Similar comment to the CAN ID's. Pass the DI channel numbers through the constructor, don't reference them directly in the header file.

File: 2024Robot_Team3484/src/main/cpp/subsystems/IntakeSubsystem.cpp
Comment (Alex-BusNet) [inline 2024Robot_Team3484/src/main/cpp/subsystems/IntakeSubsystem.cpp:48]: What is `arm_sensor_hit`? In the context of what I think this statement says, the home position sensor should never be true (thinking logically and ignoring actual normal state of sensor) when the arm is extended (unless something is wrong with the sensor) which means this statement will never run. I think what you want is:
```
if ( arm is at home )
{
    set encoder to 0 deg
    set PID setpoint to 0 deg
}
```

Alternatively, you can declare `linear_angle` as a non-const variable and set the the value of `linear_angle` to either `_intake_trapezoid.Calculate( )` _OR_ set it to 0 if the home position sensor is true, then call `_pivot_pid_controller.SetReference( )` at the end of the `Periodic( )` function.

File: 2024Robot_Team3484/src/main/cpp/commands/teleop/TeleopIntakeCommand.cpp
Comment (Alex-BusNet) [inline 2024Robot_Team3484/src/main/cpp/commands/teleop/TeleopIntakeCommand.cpp:17]: I suggest declaring changing the value passed into `SetRollerPower( )` to a constant (declared in `Constants.h`) and reference that value here. This will allow us to easily adjust the value and only have to change it in one place (especially if this value is the same for both auto and teleop commands).

File: 2024Robot_Team3484/src/main/cpp/commands/teleop/TeleopIntakeCommand.cpp
Comment (Alex-BusNet) [inline 2024Robot_Team3484/src/main/cpp/commands/teleop/TeleopIntakeCommand.cpp:33]: Same as previous comment on this file. Just multiply the constant by -1 to get the power in reverse.

File: 2024Robot_Team3484/src/main/cpp/commands/auton/AutonIntakeCommand.cpp
Comment (Alex-BusNet) [inline 2024Robot_Team3484/src/main/cpp/commands/auton/AutonIntakeCommand.cpp:13]: See comments on `TeleopIntakeCommand`

**Comment by nfoert** [comment]
We have now applied all of the feedback to the code. See [this commit](https://github.com/FRC-Team3484/X24_RobotCode/commit/4cd8a67d3bfba1946e083ab8389be68292fbedc1)

File: 2024Robot_Team3484/src/main/cpp/subsystems/IntakeSubsystem.cpp
Comment (nfoert) [reply 2024Robot_Team3484/src/main/cpp/subsystems/IntakeSubsystem.cpp]: I opted to use `constexpr` as I'm not sure if using `#define` will be able to passed as a constructor. Plus, the other constants are using `constexpr` and that makes more sense to me.

File: 2024Robot_Team3484/src/main/cpp/subsystems/IntakeSubsystem.cpp
Comment (Alex-BusNet) [inline 2024Robot_Team3484/src/main/cpp/subsystems/IntakeSubsystem.cpp:17]: Confirm naming convention on parameters with the rest of the team. I know constants are typed in all caps, but I don't believe function/constructor parameters are. The concern I have here is the all-caps designation would lead one to believe this is a constant not a parameter.

File: 2024Robot_Team3484/src/main/include/subsystems/IntakeSubsystem.h
Comment (Alex-BusNet) [inline 2024Robot_Team3484/src/main/include/subsystems/IntakeSubsystem.h:29]: Verify the datatype the PID controller expects here. I believe they should be `double` not `int` datatypes here. You can also cut down the number of parameters declared here by using `SC_PIDConstants` (you just need to include `FRC3484_Lib/utils/SC_Datatypes.h`).

**Comment by nfoert** [comment]
I have now commited the additional changes, including the motor can id naming conventions, now using `SC_PIDConstants` and have added the ports to the motors in `Constants.h`.

File: 2024Robot_Team3484/src/main/include/Constants.h
Comment (Alex-BusNet) [inline 2024Robot_Team3484/src/main/include/Constants.h:25]: You'll need the namespace qualifier here: `SC::SC_PIDConstants`

File: 2024Robot_Team3484/src/main/include/subsystems/IntakeSubsystem.h
Comment (Alex-BusNet) [inline 2024Robot_Team3484/src/main/include/subsystems/IntakeSubsystem.h:23]: You'll need the namespace qualifier here: `SC::SC_PIDConstants`

**Review by Alex-BusNet — APPROVED** [review]
PR approved. Wait for drivetrain to be pushed to be finished and merged with Development before merging this branch.

**Review by Alex-BusNet — APPROVED** [review]
As a general practice, don't push new code to a branch once its pull request has been approved. The changes I suggested here can be made in the launcher subsystem since it has merged in most of the intake code already. I suggest merging the last of the changes with that branch and then closing this PR.

**Pull request merged**
